python/cudf/cudf/core/dataframe.py

```diff
@@ -4026,16 +4017,16 @@ def stack(self, level=-1, dropna=True):
         dtype: int64
         """
         assert level in (None, -1)
-        index_as_cols = self.index.to_frame(index=False)._columns
-        new_index_cols = libcudf.filling.repeat(index_as_cols, self.shape[1])
-        [last_index] = libcudf.filling.tile(
-            [column.as_column(self.columns)], self.shape[0]
-        )
-        new_index_cols.append(last_index)
-        index_df = DataFrame(
-            dict(zip(range(0, len(new_index_cols)), new_index_cols))
+        repeated_index = self.index.repeat(self.shape[1])
+        repeated_index = [c for c in repeated_index._columns]
+
+        name_index = DataFrame({0: self._column_names}).tile(self.shape[0])
```

Comment:
Can we just use a `Frame` or a `Column` instead of a `DataFrame` here? A `DataFrame` has a lot of implicit behaviors that we would probably want to avoid if possible, i.e. materializing and tiling the `RangeIndex` unnecessarily.

Reply:
`Frame` probably works. Didn't think of that. I considered adding a method to `Column`, but thought it'd require creating a `DataFrame`. Can probably use a `Frame` instead.